mitype/commandline.py

```diff
@@ -155,3 +170,30 @@ def load_based_on_difficulty(difficulty_level=random.randrange(1, 6)):
 
     print("Select a difficulty level in range [1,5]")
     sys.exit(2)
+
+
+def show_history(N):
+    N = abs(int(N))
+    history_file = 'mitype_history.csv'
+    history_path = os.path.join(os.path.expanduser('~'), history_file)
+
+    if os.path.exists(history_path):
+        with open(history_path, 'r') as file:
+            history_reader = csv.reader(file)
+            next(history_reader)
+
+            data = list(history_reader)
+            no_of_records = len(data)
+
+            k = no_of_records if N == -1 or N >= no_of_records else N
+            print("Last " + str(k) + " records: ")
+            print("ID\tWPM\tDATE\t\tTIME")
+
+            start_count = 0 if N >= len(data) or N == -1 else no_of_records-N
+            for i in range(start_count, no_of_records):
+                print(data[i][0] + '\t' + data[i][1] + '\t' + data[i][2] + '\t' + data[i][3])
+            print()
+
+    else:
+        print("No records found!")
+        print("Take the test atleast once before viewing history.")
```

Comment:
Instead of printing this, let's create the file if it doesn't exist.

Reply:
It creates a new file if the user takes the tests once. It would be repeating the same thing _anyway_ 🙃 .

In case we create the file if it doesn't exist, do we create the file and print it is empty? That would be the same as the current thing wouldn't it?

Reply:
We don't print "Please run test once".
Instead we print "0 records found" alongwith the table structure.